packages/flame/lib/src/components/component.dart

```diff
@@ -603,7 +603,11 @@ class Component {
   /// Changes the current parent for another parent and prepares the tree under
   /// the new root.
   void changeParent(Component newParent) {
-    newParent.lifecycle._adoption.add(this);
+    if (_parent == null) {
+      addToParent(newParent);
+    } else {
+      newParent.lifecycle._adoption.add(this);
+    }
```

Comment:
What if we make this into a setter for `.parent`?

Reply:
Good idea, done